src/server/replica.cc

```diff
@@ -470,14 +470,18 @@ error_code Replica::InitiateDflySync() {
     DefaultErrorHandler(ge);
     for (auto& flow : shard_flows_)
       flow->CloseSocket();
+
+    // Reset all shard states from replica state to normal.
+    SetShardStates(false);
```

Comment:
I think that the error handler should do the unblocking tasks. This operation is more of a cleanup